src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1560,15 +1569,43 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             _terminal->SetBlockSelection(false);
             _renderer->TriggerSelection();
             _UpdateSelectionMarkersHandlers(*this, winrt::make<implementation::UpdateSelectionMarkersEventArgs>(true));
+
+            foundResults->TotalMatches(gsl::narrow<int32_t>(_searcher.Results().size()));
+            foundResults->CurrentMatch(gsl::narrow<int32_t>(_searcher.CurrentMatch()));
+
+            _terminal->AlwaysNotifyOnBufferRotation(true);
         }
 
         // Raise a FoundMatch event, which the control will use to notify
         // narrator if there was any results in the buffer
-        _FoundMatchHandlers(*this, winrt::make<implementation::FoundResultsArgs>(foundMatch));
+        _FoundMatchHandlers(*this, *foundResults);
+    }
+
+    Windows::Foundation::Collections::IVector<int32_t> ControlCore::SearchResultRows()
+    {
+        auto lock = _terminal->LockForWriting();
+        _searcher.ResetIfStale(*GetRenderData());
+
+        auto results = std::vector<int32_t>();
+
+        // use a map to remove duplicates
+        std::map<int32_t, bool> rows;
+        for (const auto& match : _searcher.Results())
+        {
+            const auto row = match.start.y;
+            // First check if it's in the map
+            if (rows.find(row) == rows.end())
+            {
+                rows[row] = true;
+                results.push_back(row);
+            }
+        }
+        return winrt::single_threaded_vector<int32_t>(std::move(results));

```

Comment:
Do we need to cache this?

Reply:
I dunno if we _need_ to but we sure can ☺️